### ISSUE TITLE
The ability to group tables based on name patterns or columns. 

### ISSUE THREAD

**Issue by Moubassher** [sqlitebrowser/sqlitebrowser#2938]
like on windows; the "group by" context menu option. Pretty useful for when a database has a large number of tables as it happens with me sometimes.

**Comment by chrisjlocke** [comment]
Wouldn't you need a higher 'group' for these to be in though?  Currently, a table only has a name field, so I assume you're wanting (imagining a stock management system) grouped by client tables, order tables, stock tables, etc.  Wouldn't an item table have to belong to multiple groups?  Its stock, but has its fingers in the orders group too,.....

**Comment by Moubassher** [comment]
aha, true. Hence my suggestion the grouping be just using name patterns,
for example. I read a large dataset from databases and took my time adding
this and other viewing options, it turned out alright. But this browser is
fantastic, I thought it would be like a cherry on top.

**Comment by mgrojo** [comment]
Maybe you could refactor that DB at the SQLite level into different files and then use the `ATTACH` statement: https://www.sqlitetutorial.net/sqlite-attach-database/

In DB4S you could save a project with all the attached databases.

**Comment by chrisjlocke** [comment]
> Hence my suggestion the grouping be just using name patterns

Aah, missed that, as it was in the title ... didn't fully read that.  👍
I can see the advantages in one way, but nine times out of ten, I find a table based on its name.  At work, we have one database but hundreds of tables based on different systems, so the tables are prefixed with that system.  It makes finding a table tricky sometimes.... 😫

**Comment by Moubassher** [comment]
NICE! I did NOT know this, just tried it. Thank you!
Should I remove the suggestion or ...? I'm kinda new here.

**Comment by chrisjlocke** [comment]
This'll work for your own application/database, but obviously won't work for another one, which would expect tables to be in specific databases... (just making sure you're aware of the limitation).

**Comment by Moubassher** [comment]
I sure know what you mean. Manuel's suggestion here pretty much did it for
me though.

**Comment by Moubassher** [comment]
of course, yes. Confession: I still do the organizing part on the frontend,
and still have a long way to go and muster that discipline to organize my
dbs right. *sigh*. You wouldn't believe the hoops I had  to jump through
just because I didn't know what should go where, junior programmer here.

**Comment by chrisjlocke** [comment]
No harm in that - we all had to learn.  That's the fun, right?  Wait until you code all the database, then decide to shuffle tables around ... then have to recode everything, only to find maybe plan A was better after all.... 😉
What are you programming in?

**Comment by Moubassher** [comment]
*nervous laugh*: not that I had to do that before or anything. And right,
it's been great so far. I'm doing JavaFX; the now-not-so-cool desktop
programming still. ;)
